pkg/client/create_manifest.go

```diff
@@ -0,0 +1,54 @@
+package client
+
+import (
+	"context"
+
+	"github.com/buildpacks/imgutil"
+)
+
+type CreateManifestOptions struct {
+	// Name of the ManifestList.
+	ManifestName string
+
+	// List of Images
+	Manifests []string
+
+	// Manifest list type (oci or v2s2) to use when pushing the list (default is v2s2)
+	MediaType imgutil.MediaTypes
+
+	// Skip creating index locally, directly publish to a registry.
+	// Requires ManifestName to be a valid registry location.
+	Publish bool
+
+	// Defines the registry to publish the manifest list.
+	Registry string
+
+	// Directory to store OCI layout
+	LayoutDir string
+}
+
+func (c *Client) CreateManifest(ctx context.Context, opts CreateManifestOptions) error {
```

Comment:
Try to add logging here, I think the client has a logger instance so you can use it, also to handle the errors.

Reply:
I will also change the signature to:

```golang

  func (c *Client) CreateManifest(ctx context.Context, name string, manifests []string, opts CreateManifestOptions) error {   }

```

And I will remove `ManifestName` and `Manifests` from `CreateManifestOptions`